src/main/java/net/glowstone/util/LibraryManager.java

```diff
@@ -113,10 +97,15 @@ public void run() {
 
             // hack it onto the classpath
             try {
-                Method method = URLClassLoader.class.getDeclaredMethod("addURL", URL.class);
-                method.setAccessible(true);
-                method.invoke(sysLoader, file.toURI().toURL());
-            } catch (ReflectiveOperationException | MalformedURLException e) {
+                String[] javaVersion = System.getProperty("java.version").split("\\.");
+                if (Integer.parseInt(javaVersion[0]) >= 9) {
+                    ClassPathAgent.addJarFile(new JarFile(file));
```

Comment:
Looks like the specific instrumentation method that ClassPathAgent uses can be used with Java 8 and below. Why not just use that there?

Reply:
But I have the problem that Laucher-Agents don't work with Java 8

Reply:
You're right, its not explicitly called out anywhere that I can see, but I can see Launcher-Agent-Class included in the doc page for [Java 9](https://docs.oracle.com/javase/9/docs/api/java/lang/instrument/package-summary.html) but not for [Java 8](https://docs.oracle.com/javase/8/docs/api/java/lang/instrument/package-summary.html).